artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/ActiveMQServerControlImpl.java

```diff
@@ -4479,5 +4482,39 @@ public void replay(String startScan, String endScan, String address, String targ
 
       server.replay(startScanDate, endScanDate, address, target, filter);
    }
+
+   @Override
+   public void stopEmbeddedWebServer() throws Exception {
+      for (ActiveMQComponent component : server.getExternalComponents()) {
```

Comment:
shouldn't this be on its own thread as well? what if you stop from the console?

don't make sense I know, but wouldn't be broken if a user do it?

Reply:
Stopping the embedded web server from the web console is not a problem. It's having the thread survive so that it can _start_ it again that's the problem.